src/Serializer/DefaultEventSerializer.php

```diff
@@ -38,10 +43,12 @@ public function serialize(object $event, array $options = []): SerializedEvent
 
     public function deserialize(SerializedEvent $data, array $options = []): object
     {
-        $class = $this->eventRegistry->eventClass($data->name);
         $payload = $this->encoder->decode($data->payload, $options);
+        $class = $this->eventRegistry->eventClass($data->name);
+
+        $upcast = ($this->upcaster)(new Upcast($class, $payload));
```

Comment:
we could change it to use the `name` instead `class`

Reply:
I think taking the name brings us more. This also allows you to rename an event without performing a migration. And the mapping still happens afterwards from name to class. So, I would want to convert it to the name.

Reply:
Updated it :)